pkg/gateway/multiparts/tracker.go

```diff
@@ -38,21 +64,28 @@ var (
 	ErrInvalidMetadataSrcFormat = errors.New("invalid metadata source format")
 )
 
-func NewTracker(adb db.Database) Tracker {
+func NewTracker(ms kv.StoreMessage) Tracker {
 	return &tracker{
-		db: adb,
+		store: ms,
 	}
 }
 
 func (m *tracker) Create(ctx context.Context, multipart MultipartUpload) error {
 	if multipart.UploadID == "" {
 		return ErrInvalidUploadID
 	}
-	_, err := m.db.Transact(ctx, func(tx db.Tx) (interface{}, error) {
-		_, err := tx.Exec(`INSERT INTO gateway_multiparts (upload_id,path,creation_date,physical_address, metadata, content_type)
-			VALUES ($1, $2, $3, $4, $5, $6)`,
-			multipart.UploadID, multipart.Path, multipart.CreationDate, multipart.PhysicalAddress, multipart.Metadata, multipart.ContentType)
-		return nil, err
+	_, err := m.Get(ctx, multipart.UploadID)
+	if err == nil {
+		return ErrAlreadyExists
+	}
+
+	err = m.store.SetMsg(ctx, []byte(multipart.UploadID), &MultipartUploadData{
```

Comment:
We need hierarchy for the keys in the store. Imagine creating a `Branch` and an `Upload` with the same ID.

Reply:
I think this should be solved by making sure we create a unique hash (UUID / GUID) every time. Creating a conflicting BranchID / UploadID / RefID is a bug in my opinion